File: dataset/java2python/java/CONSTRUCT_LEXICOGRAPHICALLY_SMALLEST_PALINDROME.java
String f_gold ( char [ ] str , int len ) {
  int i = 0 , j = len - 1 ;
  for ( ;
  i < j ;
  i ++ , j -- ) {
    if ( str [ i ] == str [ j ] && str [ i ] != '*' ) continue ;
    else if ( str [ i ] == str [ j ] && str [ i ] == '*' ) {
      str [ i ] = 'a' ;
      str [ j ] = 'a' ;
      continue ;
    }
    else if ( str [ i ] == '*' ) {
      str [ i ] = str [ j ] ;
      continue ;
    }
    else if ( str [ j ] == '*' ) {
      str [ j ] = str [ i ] ;
      continue ;
    }
    System . out . println ( "Not Possible" ) ;
    return "" ;
  }
  return String . valueOf ( str ) ;
}

File: result/translate/starchat/java2python/python/CONSTRUCT_LEXICOGRAPHICALLY_SMALLEST_PALINDROME.py
def f_gold(str, len):
        i = 0
        j = len - 1
        while i < j:
            if str[i] == str[j] and str[i]!= '*':
                i += 1
                j -= 1
            elif str[i] == str[j] and str[i] == '*':
                str[i] = 'a'
                str[j] = 'a'
                i += 1
                j -= 1
            elif str[i] == '*':
                str[i] = str[j]
                i += 1
            elif str[j] == '*':
                str[j] = str[i]
                j -= 1
            else:
                print("Not Possible")
                return ""
        return ''.join(str)